src/drag_drop/table/Table.js

```diff
@@ -0,0 +1,57 @@
+import React, { useRef } from 'react';
+import { AgGridReact } from 'ag-grid-react';
+
+import 'ag-grid-community/dist/styles/ag-grid.css';
+import 'ag-grid-community/dist/styles/ag-theme-alpine-dark.css';
+
+function GenerateTable(props) {
+
+  const gridRef = useRef();
+
+  const robots = ['R2-D2', 'C-3PO', 'BB-8', 'L3-37', 'K-2SO', 'IG-11']; // replace with backend endpoint
+
+  const defaultColDef = {
+    editable: true,
+    resizable: true
+  }
+
+  const columnDefs = [
+      { field: 'ID', rowDrag: true, editable: false },
+      { field: 'x' },
+      { field: 'y' },
+      { field: 'z' },
+      { field: 'roll' },
+      { field: 'pitch' },
+      { field: 'yaw' },
+      { field: 'robot', cellEditor: 'agSelectCellEditor', singleClickEdit: true, cellEditorParams: { values: robots } }
+  ];
+
+  const rowData = props.data;
+
+  function handleOnSubmit(){
+    console.log(gridRef.current.api.getDataAsCsv());  // replace with backend endpoint that provides error message for invalid input
```

Comment:
Even this waits for the backend integration:

See above: Don't use refs.
See below: Suggestion for getting this done using the library's own API without refs.

Reply:
documentation suggests it with refs. https://ag-grid.com/react-data-grid/csv-export/

Reply:
I see less of a suggestion and more of an example on the `ag-grid-react` docs. And where examples go against `react` docs I'd prefer deviating from the examples